controllers/service/service_controller.go

```diff
@@ -184,7 +187,7 @@ func (r *serviceReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manag
 
 func (r *serviceReconciler) setupWatches(_ context.Context, c controller.Controller) error {
 	svcEventHandler := eventhandlers.NewEnqueueRequestForServiceEvent(r.eventRecorder, r.annotationParser,
-		r.logger.WithName("eventHandlers").WithName("service"))
+		r.logger.WithName("eventHandlers").WithName("service"), r.defaultTargetType != "")
```

Comment:
this will create a confusion to the end users. If they happen to enable this setting without realizing the kcm needs to be disabled, then both kcm and this controller would end up provisioning lb for the same service.

Reply:
I'm failing to understand the relevance of the comment to the context that it was made in.

Assuming it's a comment about the feature as a whole, it is inherent to the feature of being able to act as the default service controller that no other thing in the cluster can also be the default service controller at the same time.